### ISSUE TITLE
Request for adding Encrypted Client Hello (ECH) to improve privacy 

### ISSUE THREAD

**Issue by malaguer** [tuyafeng/Via#1473]
Please, could you add Encrypted Client Hello protocol to Via browser as an option?

This standard is already working for Chromium based browsers and Firefox through its internal flags and settings.

Here you have an explanation on Cloudfare blog explaining how it works: [Encrypted Client Hello - the last puzzle piece to privacy](https://blog.cloudflare.com/announcing-encrypted-client-hello).

Actually I'm not sure if this is possible to do in a Webview based browser but I would like you to consider this option as a future enhancement.

Thanks in advance and congrats for such an amazing and light browser!

**Comment by malaguer** [comment]
After some investigation, I would like to clarify the scenario for which I was requesting the implementation of ECH, in case someone finds themselves in a similar situation. I've encountered an effective solution. 

I had peoblems to access certain websites blocked by arbitrary restrictions imposed by ISPs in my country. The issue, I discovered, was related to Deep Packet Inspection (DPI).

To bypass packet inspection on Android I'm using now [AdGuard](https://github.com/AdguardTeam/AdguardForAndroid) in combination with [PowerTunnel](https://github.com/krlvm/PowerTunnel-Android?tab=readme-ov-file) app. AdGuard in VPN mode, which I already used to block ads in other apps, and PowerTunnel with its addon LibertyTunnel, which does exactly what I need and successfully circumvents the blocking of problematic websites.

To use both apps simultaneously, simply use PowerTunnel in local proxy mode and configure AdGuard to use this proxy for Via Browser or any app experiencing ISP blocking. Everything now functions as intended.

I hope this information is useful for anyone facing similar issues with their ISP.
